source/extensions/transport_sockets/alts/alts_proxy.cc

```diff
@@ -0,0 +1,153 @@
+#include "source/extensions/transport_sockets/alts/alts_proxy.h"
+
+#include <memory>
+#include <utility>
+
+#include "absl/memory/memory.h"
+#include "absl/status/status.h"
+#include "absl/strings/string_view.h"
+#include "absl/types/span.h"
+#include "src/proto/grpc/gcp/handshaker.pb.h"
+#include "src/proto/grpc/gcp/transport_security_common.pb.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace TransportSockets {
+namespace Alts {
+
+using ::grpc::gcp::HandshakeProtocol;
+using ::grpc::gcp::HandshakerReq;
+using ::grpc::gcp::HandshakerResp;
+using ::grpc::gcp::HandshakerService;
+using ::grpc::gcp::NextHandshakeMessageReq;
+using ::grpc::gcp::ServerHandshakeParameters;
+using ::grpc::gcp::StartClientHandshakeReq;
+using ::grpc::gcp::StartServerHandshakeReq;
+
+void AltsProxy::setRpcProtocolVersions(grpc::gcp::RpcProtocolVersions* rpc_protocol_versions) {
+  rpc_protocol_versions->mutable_max_rpc_version()->set_major(MaxMajorRpcVersion);
+  rpc_protocol_versions->mutable_max_rpc_version()->set_minor(MaxMinorRpcVersion);
+  rpc_protocol_versions->mutable_min_rpc_version()->set_major(MinMajorRpcVersion);
+  rpc_protocol_versions->mutable_min_rpc_version()->set_minor(MinMinorRpcVersion);
+}
+
+absl::StatusOr<std::unique_ptr<AltsProxy>>
+AltsProxy::create(std::shared_ptr<grpc::Channel> handshaker_service_channel) {
+  if (handshaker_service_channel == nullptr) {
+    return absl::InvalidArgumentError("Handshaker service channel is null.");
+  }
+  auto client_context = std::make_unique<grpc::ClientContext>();
```

Comment:
Do we need to set_deadline for the gRPC client through the ClientContext, since it is possible to end up waiting forever?

Reply:
Good idea, done.